### ISSUE TITLE
No valid gcc optimizations

### ISSUE THREAD

**Issue by jserv** [jserv/xv6-x86_64#5]
Using `make OPT=-O1` specifies the optimization order of gcc. However, it always fails for O=1, 2, and s.

The case of -O1:
- build FAIL

```
nmeta 29 (boot, super, inode blocks 26, bitmap blocks 1) blocks 941 log 30 total 1000
mkfs: tools/mkfs.c:268: iappend: Assertion `fbn < (10 + (512 / sizeof(uint)))' failed.
Makefile:221: recipe for target 'fs.img' failed
```

The case of -O2:
- build FAIL

```
nmeta 29 (boot, super, inode blocks 26, bitmap blocks 1) blocks 941 log 30 total 1000
mkfs: tools/mkfs.c:268: iappend: Assertion `fbn < (10 + (512 / sizeof(uint)))' failed.
Makefile:221: recipe for target 'fs.img' failed
```

The case of -Os:
- build OK
- `usertests` hangs:

```
...
open small succeeded ok
read succeeded ok
small file test ok
big files test
```

**Comment by jserv** [comment]
Fixed in commit 5cdf06abedd626c803e0ece674821e8f842ea254

However, we should introduce an efficient way to reduce the whole size of user programs.

**Comment by Yo-yo-ooo** [comment]
But I.....
<img width="931" alt="image" src="https://github.com/jserv/xv6-x86_64/assets/98507772/faaaf4b1-92ad-4076-b5ff-3eb95e0cc272">
